go/teams/service_helper.go

```diff
@@ -669,85 +669,63 @@ func InviteEmailPhoneMember(ctx context.Context, g *libkb.GlobalContext, teamID
-				continue
-			}
-
-			// api server side of this only accepts x.yy domain name:
-			parts := strings.Split(addr.Address, ".")
-			if len(parts[len(parts)-1]) < 2 {
-				g.Log.CDebugf(ctx, "team %s: skipping malformed email (domain) %q: %s", teamname, email, parseErr)
-				res.Malformed = append(res.Malformed, email)
-				continue
-			}
-
-			name := keybase1.TeamInviteName(addr.Address)
-			existing, err := t.HasActiveInvite(libkb.NewMetaContext(ctx, g), name, "email")
-			if err != nil {
-				return err
-			}
-			if existing {
-				g.Log.CDebugf(ctx, "team %s: invite for %s already exists, omitting from invite list",
-					teamname, name)
-				res.AlreadyInvited = append(res.AlreadyInvited, addr.Address)
-				continue
-			}
-			inv := SCTeamInvite{
-				Type: "email",
-				Name: name,
-				ID:   NewInviteID(),
-			}
-			invites = append(invites, inv)
-			res.Invited = append(res.Invited, addr.Address)
+		// api server side of this only accepts x.yy domain name:
+		parts := strings.Split(addr.Address, ".")
+		if len(parts[len(parts)-1]) < 2 {
+			g.Log.CDebugf(ctx, "team %s: skipping malformed email (domain) %q", teamname, email)
+			res.Malformed = append(res.Malformed, email)
+			continue
 		}
-		if len(invites) == 0 {
-			g.Log.CDebugf(ctx, "team %s: after existing filter, no one to invite", teamname)
-			// return value assign to escape closure
-			resOuter = res
-			return nil
+		a, parseErr := libkb.ParseAssertionURLKeyValue(g.MakeAssertionContext(mctx), "email", addr.Address, false)
+		if parseErr != nil {
+			g.Log.CDebugf(ctx, "team %q: skipping malformed email %q; could not parse into assertion: %s", teamname, email, parseErr)
+			res.Malformed = append(res.Malformed, email)
+			continue
 		}
+		toAdd = append(toAdd, keybase1.UserRolePair{AssertionOrEmail: a.String(), Role: role})
+		emailsParsed = append(emailsParsed, addr.Address)
+	}
 
-		var teamInvites SCTeamInvites
-		switch role {
-		case keybase1.TeamRole_ADMIN:
-			teamInvites.Admins = &invites
-		case keybase1.TeamRole_WRITER:
-			teamInvites.Writers = &invites
-		case keybase1.TeamRole_READER:
-			teamInvites.Readers = &invites
-		case keybase1.TeamRole_OWNER:
-			teamInvites.Owners = &invites
-		default:
-			return fmt.Errorf("unknown team role: %s", role)
-		}
+	if len(toAdd) == 0 {
+		return res, err
+	}
 
-		g.Log.CDebugf(ctx, "team %s: after existing filter, inviting %d emails as %s", teamname, len(invites), role)
-		err = t.postTeamInvites(ctx, teamInvites)
+	err = RetryIfPossible(ctx, g, func(ctx context.Context, _ int) error {
+
+		t, err := GetForTeamManagementByStringName(ctx, g, teamname, true)
 		if err != nil {
 			return err
 		}
-		// return value assign to escape closure
-		resOuter = res
-		return nil
+
+		// TRIAGE-1864 to handle this output better in case of errors
+		_, _, err = AddMembers(ctx, g, t.ID, toAdd)
+
+		return err
 	})
-	return resOuter, err
+
+	if err != nil {
+		return res, err
+	}
+
+	res.Invited = emailsParsed
```

Comment:
this isn't quite right, right? since some of those emails could have resolved? and before we were populating AlreadyInvited too. You could just delete these fields from the AVDL if we're not using them anywhere.